integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,139 @@
+            raise
+
+    @staticmethod
+    def parse_projects_from_folders(folders: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        folder_with_projects = [folder["lists"] for folder in folders]
+        flattened_projects = [project for sublist in folder_with_projects for project in sublist]
+        return flattened_projects
+
+    @staticmethod
+    def _generate_base_req_params(page: int = 0) -> dict[str, Any]:
+        return {"page": page}
+
+    async def _get(self, endpoint: str, params: dict[str, Any] = None) -> dict[str, Any]:
+        url = f"{self.api_url}{endpoint}"
+        response = await self.client.get(url, params=params)
+        response.raise_for_status()
+        return response.json()
+
+    async def _get_teams(self) -> dict[str, Any]:
+        return await self._get("/team")
+
+    async def _get_spaces(self, team_id: str) -> dict[str, Any]:
+        return await self._get(f"/team/{team_id}/space")
+
+    async def _get_tasks(self, list_id: str, params: dict[str, Any]) -> dict[str, Any]:
+        return await self._get(f"/list/{list_id}/task", params)
+
+    async def _get_folderless_projects(self, space_id: str) -> dict[str, Any]:
+        return await self._get(f"/space/{space_id}/list")
+
+    async def _get_folders(self, space_id: str) -> dict[str, Any]:
+        return await self._get(f"/space/{space_id}/folder")
+
+    async def get_projects(self, space_id: str, team_id: str) -> list[dict[str, Any]]:
+        logger.info("Getting projects from ClickUp")
+
+        try:
+            foldered_projects = self.parse_projects_from_folders(
+                (await self._get_folders(space_id))["folders"]
+            )
+            folderless_projects = (await self._get_folderless_projects(space_id))["lists"]
+            projects = folderless_projects + foldered_projects
+
+            if not projects:
+                logger.warning("Space query returned 0 projects")
+
+            return [{**project, "__team_id": team_id} for project in projects]
+        except httpx.HTTPStatusError as e:
+            logger.error(f"Failed to fetch projects for space_id {space_id}: {e}")
+            return []
+
+    async def fetch_all_projects(self):
+        teams = await self.get_teams()
+        for team in teams:
+            team_id = team["id"]
+            spaces = await self.get_spaces(team_id)
+            logger.info(f"Received spaces batch with {len(spaces)} spaces for team {team_id}")
+            for space in spaces:
+                space_id = space["id"]
+                projects = await self.get_projects(space_id, team_id)
+                return projects, team_id
+
+    async def get_teams(self) -> list[dict[str, Any]]:
+        logger.info("Getting teams from ClickUp")
+        try:
+            total_teams = (await self._get_teams())["teams"]
+            return total_teams
+        except httpx.HTTPStatusError as e:
+            logger.error(f"Failed to fetch teams: {e}")
+            return []
```

Comment:
What's the point of have a _get_teams() and get_teams() they are both doing the same thing. This is means you don't need this function. Just change the _get_teams, _get_spaces etc directly in the main.py of you need to rename them so they don't have the underscores in front of them